### ISSUE TITLE
Is there a detector that detects the problem of synchronized on local variables?

### ISSUE THREAD

**Issue by guoxiangbin** [spotbugs/spotbugs#590]
As shown in the following test cases:

```
public void func1() {
	Object lock = new Object();

        /* Should report a bug */
	synchronized (lock) {
		i++;
	}
}
```
